annotation-utils/src/main/java/eu/europeana/annotation/utils/serialize/AnnotationLdSerializer.java

```diff
@@ -142,30 +145,29 @@ protected void putTarget(Annotation annotation, JsonLdResource jsonLdResource) {
 
 	protected void putBody(Annotation annotation, JsonLdResource jsonLdResource) {
 		//if (!annotation.getInternalType().equals(AnnotationTypes.OBJECT_LINKING.name())) {
-		if (annotation.getBody()!= null) {
-			// tag or comment, not linking
-			if (isJsonObjectInput(annotation.getBody().getInputString())) {
-				// annotation.getBody().setInputString(null);
-				JsonLdProperty bodyProperty = addBodyProperty(annotation);
-				if (bodyProperty != null)
-					jsonLdResource.putProperty(bodyProperty);
-			} else {
-				if (annotation.getBody().getValues() != null && annotation.getBody().getValues().size() > 0) {
-					putListProperty(WebAnnotationFields.BODY, annotation.getBody().getValues(), jsonLdResource, true);
-				} else{
-				    	//TODO: check if this is still correct
-				    	//if no values in the body, 
-				    	//linking  have not
-				    	if(annotation.getTarget().getInputString() != null) {
-						putStringProperty(WebAnnotationFields.BODY, annotation.getBody().getInputString(), jsonLdResource);
-//						jsonLdResource.putProperty(WebAnnotationFields.BODY, annotation.getBody().getInputString());
-				    	} else {
-						putStringProperty(WebAnnotationFields.BODY, annotation.getBody().getHttpUri(), jsonLdResource);
-//						jsonLdResource.putProperty(WebAnnotationFields.BODY, annotation.getBody().getHttpUri());
-				    	}
-				}
+		// tag or comment, not linking
```

Comment:
is there a reason to remove the not null check? Even if not called now for linking, it is better to prevent NPE when this method if missused in the future

Reply:
The body null check is done upon the annotation verification which always happens before the serialization, so this check is not necessary.